### PR TITLE
HDDS-1500 : Allocate block failures in client should print exception trace.

### PR DESCRIPTION
**Pull request by avijayanhwx**
Minor change.

### CODE REVIEW

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java [hunk 1]
```diff
@@ -297,7 +297,8 @@ BlockOutputStreamEntry allocateBlockIfNeeded() throws IOException {
         succeededAllocates += 1;
       } catch (IOException ioe) {
         LOG.error("Try to allocate more blocks for write failed, already "
-            + "allocated " + succeededAllocates + " blocks for this write.");
+            + "allocated " + succeededAllocates + " blocks for this write.",
```
Comment (jiwq) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java]:
```suggestion
            + "allocated {} blocks for this write.", succeededAllocates, ioe);
```

File: hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java [hunk 2]
```diff
@@ -297,7 +297,8 @@ BlockOutputStreamEntry allocateBlockIfNeeded() throws IOException {
...
+            ioe);
```
Comment (jiwq) [inline hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/BlockOutputStreamEntryPool.java]:
```suggestion
```

**Comment by avijayanhwx** [comment]
Thank you for the suggestion @jiwq

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 52 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 457 | trunk passed |
| +1 | compile | 210 | trunk passed |
| +1 | checkstyle | 51 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 829 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 124 | trunk passed |
| 0 | spotbugs | 267 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 468 | trunk passed |
||| _ Patch Compile Tests _ |
| -1 | mvninstall | 56 | hadoop-ozone in the patch failed. |
| -1 | compile | 35 | hadoop-ozone in the patch failed. |
| -1 | javac | 35 | hadoop-ozone in the patch failed. |
| -0 | checkstyle | 16 | The patch fails to run checkstyle in hadoop-ozone |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 655 | patch has no errors when building and testing our client artifacts. |
| -1 | javadoc | 35 | hadoop-ozone in the patch failed. |
| -1 | findbugs | 76 | hadoop-ozone in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 168 | hadoop-hdds in the patch failed. |
| -1 | unit | 46 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 3929 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/801 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux dfd6d755f3f2 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3418bbb |
| Default Java | 1.8.0_191 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-mvninstall-hadoop-ozone.txt |
| compile | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-compile-hadoop-ozone.txt |
| javac | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-compile-hadoop-ozone.txt |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out//home/jenkins/jenkins-slave/workspace/hadoop-multibranch_PR-801/out/maven-patch-checkstyle-hadoop-ozone.txt |
| javadoc | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-javadoc-hadoop-ozone.txt |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-findbugs-hadoop-ozone.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/testReport/ |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/client U: hadoop-ozone/client |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 395 | trunk passed |
| +1 | compile | 205 | trunk passed |
| +1 | checkstyle | 49 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 752 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 117 | trunk passed |
| 0 | spotbugs | 232 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 411 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 379 | the patch passed |
| +1 | compile | 202 | the patch passed |
| +1 | javac | 202 | the patch passed |
| +1 | checkstyle | 55 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 591 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 121 | the patch passed |
| +1 | findbugs | 428 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 134 | hadoop-hdds in the patch failed. |
| -1 | unit | 1492 | hadoop-ozone in the patch failed. |
| +1 | asflicense | 39 | The patch does not generate ASF License warnings. |
| | | 5535 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.client.rpc.TestOzoneRpcClientWithRatis |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-801/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/801 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ee90bb429995 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3418bbb |
| Default Java | 1.8.0_191 |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/3/artifact/out/patch-unit-hadoop-hdds.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/3/artifact/out/patch-unit-hadoop-ozone.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/3/testReport/ |
| Max. process+thread count | 5405 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/client U: hadoop-ozone/client |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-801/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by avijayanhwx** [comment]
\label ozone

**Comment by mukul1987** [comment]
Thanks for the updated patch @avijayanhwx . I will commit this now.

**Pull request merged**